ProduceOnlyReferenceAssembly=true fails with `Do not use refout when using refonly`

The following project fails to compile:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <ProduceOnlyReferenceAssembly>true</ProduceOnlyReferenceAssembly>
  </PropertyGroup>

</Project>
```
The above project will fail to compile with error
> CSC : error CS8308: Do not use refout when using refonly.

This used to work with target `netstandard2.0` instead.

The workaround is to set the following to false:

```xml
    <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
```

However it's rather counter-intuitive that to generate a reference assembly - but __only__ a reference assembly - you actually have to turn off producing a reference assembly.

It is likely related to this bit and explains why netstandard2.0 works but net5.0+ doesn't:
https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L233

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets
@@ -230,7 +230,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup>
-    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and ('$(ProduceOnlyReferenceAssembly)' == '' or '$(ProduceOnlyReferenceAssembly)' == 'false')" >true</ProduceReferenceAssembly>
+    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and ('$(ProduceOnlyReferenceAssembly)' != 'true')" >true</ProduceReferenceAssembly>
   </PropertyGroup>
 
   <!--

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets
@@ -230,7 +230,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup>
-    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))" >true</ProduceReferenceAssembly>
+    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and ('$(ProduceOnlyReferenceAssembly)' == '' or '$(ProduceOnlyReferenceAssembly)' == 'false')" >true</ProduceReferenceAssembly>
   </PropertyGroup>
 
   <!--

--- a/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs
+++ b/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs
@@ -860,5 +860,62 @@ class Program
                 .Pass();
         }
 
+        [Fact]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_set()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "MainProject",
+                TargetFrameworks = "net5.0",
+                IsSdkProject = true,
+                IsExe = true
+
+            };
+
+            testProject.AdditionalProperties["ProduceOnlyReferenceAssembly"] = "true";
+
+            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testProjectInstance);
+            buildCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var outputPath = buildCommand.GetOutputDirectory(targetFramework: "net5.0").FullName;
+            var refPath = Path.Combine(outputPath, "ref");
+            Directory.Exists(refPath)
+                .Should()
+                .BeFalse();
+        }
+
+        [Fact]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_false()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "MainProject",
+                TargetFrameworks = "net5.0",
+                IsSdkProject = true,
+                IsExe = true
+
+            };
+
+            testProject.AdditionalProperties["ProduceOnlyReferenceAssembly"] = "false";
+
+            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testProjectInstance);
+            buildCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var outputPath = buildCommand.GetOutputDirectory(targetFramework: "net5.0").FullName;
+            var refPath = Path.Combine(outputPath, "ref", "MainProject.dll");
+            File.Exists(refPath)
+                .Should()
+                .BeTrue();
+        }
     }
 }

--- a/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs
+++ b/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs
@@ -860,8 +860,10 @@ class Program
                 .Pass();
         }
 
-        [Fact]
-        public void It_builds_the_project_successfully_with_only_reference_assembly_set()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_set(bool produceOnlyReferenceAssembly)
         {
             var testProject = new TestProject()
             {
@@ -869,12 +871,11 @@ class Program
                 TargetFrameworks = "net5.0",
                 IsSdkProject = true,
                 IsExe = true
-
             };
 
-            testProject.AdditionalProperties["ProduceOnlyReferenceAssembly"] = "true";
+            testProject.AdditionalProperties["ProduceOnlyReferenceAssembly"] = produceOnlyReferenceAssembly.ToString();
 
-            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject);
+            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject, identifier: produceOnlyReferenceAssembly.ToString());
 
             var buildCommand = new BuildCommand(testProjectInstance);
             buildCommand
@@ -883,39 +884,20 @@ class Program
                 .Pass();
 
             var outputPath = buildCommand.GetOutputDirectory(targetFramework: "net5.0").FullName;
-            var refPath = Path.Combine(outputPath, "ref");
-            Directory.Exists(refPath)
-                .Should()
-                .BeFalse();
-        }
-
-        [Fact]
-        public void It_builds_the_project_successfully_with_only_reference_assembly_false()
-        {
-            var testProject = new TestProject()
+            if (produceOnlyReferenceAssembly == true)
             {
-                Name = "MainProject",
-                TargetFrameworks = "net5.0",
-                IsSdkProject = true,
-                IsExe = true
-
-            };
-
-            testProject.AdditionalProperties["ProduceOnlyReferenceAssembly"] = "false";
-
-            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject);
-
-            var buildCommand = new BuildCommand(testProjectInstance);
-            buildCommand
-                .Execute()
-                .Should()
-                .Pass();
-
-            var outputPath = buildCommand.GetOutputDirectory(targetFramework: "net5.0").FullName;
-            var refPath = Path.Combine(outputPath, "ref", "MainProject.dll");
-            File.Exists(refPath)
-                .Should()
-                .BeTrue();
+                var refPath = Path.Combine(outputPath, "ref");
+                Directory.Exists(refPath)
+                    .Should()
+                    .BeFalse();
+            }
+            else
+            {
+                var refPath = Path.Combine(outputPath, "ref", "MainProject.dll");
+                File.Exists(refPath)
+                    .Should()
+                    .BeTrue();
+            }
         }
     }
 }

